src/Common/CMakeLists.txt

```diff
@@ -1,48 +1,50 @@
-project(CemuCommon)
-
-file(GLOB CPP_FILES *.cpp)
-file(GLOB H_FILES *.h)
-add_library(CemuCommon ${CPP_FILES} ${H_FILES})
+add_library(Common
+  betype.h
+  enumFlags.h
+  ExceptionHandler/ExceptionHandler.cpp
+  ExceptionHandler/ExceptionHandler.h
+  filestream.h
+  GLInclude/glext.h
+  GLInclude/glFunctions.h
+  GLInclude/GLInclude.h
+  GLInclude/glxext.h
+  GLInclude/khrplatform.h
+  GLInclude/wglext.h
+  unix/date.h
+  unix/fast_float.h
+  MemPtr.h
+  platform.h
+  precompiled.cpp
+  precompiled.h
+  socket.h
+  StackAllocator.h
+  SysAllocator.cpp
+  SysAllocator.h
+  version.h
+  zstring_view.h
+)
 
-set_property(TARGET CemuCommon PROPERTY MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
+target_precompile_headers(Common PUBLIC precompiled.h)
 
-if(WIN32)
-target_sources(CemuCommon
-PRIVATE
-    windows/platform.cpp
-    windows/platform.h
-)
+if (WIN32)
+  target_sources(Common PRIVATE windows/platform.cpp windows/platform.h)
 else()
-target_sources(CemuCommon
-PRIVATE
-    unix/platform.cpp
-    unix/platform.h
-)
+  target_sources(Common PRIVATE unix/platform.cpp unix/platform.h)
 endif()
 
-target_sources(CemuCommon
-        PRIVATE
-        ExceptionHandler/ExceptionHandler.cpp
-        ExceptionHandler/ExceptionHandler.h
-        )
 
-target_precompile_headers(CemuCommon PUBLIC precompiled.h)
-target_include_directories(CemuCommon PUBLIC "../")
+target_link_libraries(Common PRIVATE Cafe Config Components)
 
-target_link_libraries(CemuCommon PRIVATE
-	CemuCafe
-	CemuConfig
-	CemuComponents
-	Boost::nowide
-	Boost::filesystem
-	glm::glm
+target_link_libraries(Common PRIVATE
+  Boost::nowide
+  Boost::filesystem
+  Boost::headers
+  glm::glm
+  fmt::fmt
```

Comment:
Why were `Boost::headers` and `fmt::fmt` moved from `PUBLIC` to `PRIVATE?

Reply:
I don't understand what you mean. What were you trying to use `target_include_directories()` for?

Reply:
To add the headers that were previously added by making `Boost::headers` and `fmt::fmt:` `PUBLIC`.

Reply:
You shouldn't use `target_include_directories()` for that. `Boost::headers` and `fmt::fmt` are "imported targets". They are not actual libraries, but are a like logical container of libraries, headers, flags, etc. They represent a dependency.

The way CMake decided that you should make use of these imported targets is by using the `target_link_libraries()` option. When you do, for example, `target_link_libraries(MyTarget PRIVATE Boost::headers)` what actually happens it that CMake will look at the contents of `Boost::headers` and add `-I/usr/include/boost-1.74` to the list of flags of MyTarget.

If you don't know what these functions really do, please take a look at CMake's documentation; it'll make your life easier:

- https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#pseudo-targets
- https://cmake.org/cmake/help/latest/command/target_link_libraries.html

Reply:
Ah, that makes sense now. Thanks